db/db_creator.py

```diff
@@ -41,33 +41,60 @@ def __init__(
         )
         self.engine: Engine = create_engine(self.conn_str)
 
-        self.append_table_names: List[str] = append_table_names
-        self.append_table_names += DEFAULT_APPEND_TABLE_NAMES
-
     def get_table_names(self) -> List[str]:
+        '''
+        Gets table names using the SQLAlchemy Engine object.
+        '''
         logger.debug('get_table_names')
         return self.engine.table_names()
 
     def migrate(self) -> DBCreator:
+        '''
+        Updates database schema using yoyo-migrations and the migration files in the
+        migrations directory collocated with this file (db_creator.py).
+        '''
         logger.debug('migrate')
         backend = get_backend(self.conn_str)
         migrations = read_migrations(MIGRATIONS_PATH)
         with backend.lock():
             backend.apply_migrations(backend.to_apply(migrations))
         return self
 
-    def drop_records(self) -> DBCreator:
+    def drop_records(self, spec_table_names: Union[Sequence[str], None] = None) -> DBCreator:
+        '''
+        Drops records from either the specified database tables or all of the
+        application-managed database tables; tables managed by yoyo-migrations are ignored.
+        '''
         logger.debug('drop_records')
+        app_table_names = [
+            table_name for table_name in self.get_table_names() if 'yoyo' not in table_name
+        ]
+        logger.debug(f'app_table_names: {app_table_names}')
+
+        if spec_table_names is None:
+            # Drop all non-yoyo tables
+            drop_table_names = app_table_names
+        else:
+            # Drop specified table names if they're valid (i.e. an application-managed table)
+            drop_table_names = []
+            for spec_table_name in spec_table_names:
+                if spec_table_name in app_table_names:
+                    drop_table_names.append(spec_table_name)
+                else:
+                    logger.error(f'Invalid table name was provided: {spec_table_name}')
+
         conn = self.engine.connect()
         conn.execute('SET FOREIGN_KEY_CHECKS=0;')
-        for table_name in self.get_table_names():
-            if 'yoyo' not in table_name and table_name not in self.append_table_names:
-                logger.debug(f'Table Name: {table_name}')
-                conn.execute(f'DELETE FROM {table_name};')
-                logger.info(f'Dropped records in {table_name} in {self.db_name}')
+        for drop_table_name in drop_table_names:
+            logger.debug(f'Table Name: {drop_table_name}')
+            conn.execute(f'DELETE FROM {drop_table_name};')
```

Comment:
Looking into the `sqlalchemy` library, the `execute` function will raise an exception ( in the unknown situation). This should be wrapped in try-catch. Similarly, in the other parts of code that are doing a DB transaction. This issue need not be solved here but can be taken as a separate issue

Reply:
Yeah, was thinking about this a little. I'll do some testing and can resolve it here.

Reply:
@pushyamig, okay, I was a little concerned about us trying to delete records from tables that don't exist when using `reset_database`, but because I'm relying on `self.engine.table_names`, we're only ever going to get actual tables passed to the `DELETE` query. When specific strings are passed, I'm doing validation.

Can you think of other cases where we would really need to catch an exception? `try/except` can be useful, but I prefer to not use unless necessary, since it can lead to some confusing output if the exceptions aren't specific enough.

Reply:
https://github.com/tl-its-umich-edu/course-inventory/pull/129/files#r419535834